### PR TITLE
acceptance/cluster: correctly detect deadline exceeded errors

### PR DESCRIPTION
**Pull request by petermattis**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5781)

### CODE REVIEW

File: acceptance/cluster/docker.go [hunk 1]
```diff
@@ -358,7 +358,10 @@ func (cli retryingDockerClient) retry(ctx context.Context, timeout time.Duration
 	for i := 0; i < cli.attempts; i++ {
 		timeoutCtx, _ := context.WithTimeout(ctx, timeout)
 		err := f(timeoutCtx)
-		if err != context.DeadlineExceeded {
+		// docker-engine/client wraps the context.DeadlineExceeded with its own
+		// error message, forcing us to detect deadline exceeded by string
+		// matching.
+		if !strings.Contains(err.Error(), context.DeadlineExceeded.Error()) {
```
Comment (petermattis) [inline acceptance/cluster/docker.go]: @tamird I might be able to look at `timeoutCtx.Err()`, but let's just stick with this for now as I know the "deadline exceeded" messages is being output.

**Comment by tamird** [comment]
:+1:

File: acceptance/cluster/docker.go [hunk 2]
```diff
@@ -358,7 +358,12 @@ func (cli retryingDockerClient) retry(ctx context.Context, timeout time.Duration
 	for i := 0; i < cli.attempts; i++ {
 		timeoutCtx, _ := context.WithTimeout(ctx, timeout)
 		err := f(timeoutCtx)
-		if err != context.DeadlineExceeded {
+		// docker-engine/client wraps the context.DeadlineExceeded with its own
```
Comment (tschottdorf) [inline acceptance/cluster/docker.go]: so silly.

**Comment by tschottdorf** [comment]
looks bueno to me

**Comment by BramGruneir** [comment]
LGTM

**Pull request merged**